libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -75,18 +75,14 @@ struct CopyTranslate: public yul::ASTCopier
 			{
 				solAssert(reference.isOffset != reference.isSlot, "");
 
-				pair<u256, unsigned> slot_offset = m_context.storageLocationOfVariable(*varDecl);
-
-				string const value = reference.isSlot ?
-					slot_offset.first.str() :
-					to_string(slot_offset.second);
-
-				return yul::Literal{
-					_identifier.location,
-					yul::LiteralKind::Number,
-					yul::YulString{value},
-					{}
-				};
+				auto [slot, offset] = m_context.storageLocationOfVariable(*varDecl);
+
+				string const value = reference.isSlot ? slot : offset;
+
+				if (varDecl->isStateVariable() || (!reference.isSlot && varDecl->type()->stackItems().size() == 1))
```

Comment:
What about checking if it starts with `[0-9]` (works for both decimals and hex)?